### ISSUE TITLE
binja: optimize performance of name demangling

### ISSUE THREAD

**Issue by williballenthin** [mandiant/capa#1610]
https://github.com/mandiant/capa/blob/430f9da449cbd2c7142594f3bcf370cbcfee37d8/capa/features/extractors/binja/helpers.py#L46

**Comment by tehami02** [comment]
@williballenthin I want to work on this.

**Comment by williballenthin** [comment]
this issue requires a license of Binary Ninja in order to triage and develop. do you have one?

you'll also need to work with @xusheng6 to clearly define the problem and propose a fix. i recall chatting with him about an opportunity for better performance but i don't remember the details.

**Comment by tehami02** [comment]
@williballenthin No I dont have Binary Ninja License, maybe I should try some GFI's first.

**Comment by williballenthin** [comment]
i think #1607 and #1608 might be reasonable places to start.

**Comment by tehami02** [comment]
@williballenthin Thanks for reference, I will try them.

**Comment by tehami02** [comment]
@williballenthin I need little help, I actually forked capa repo and cloned it into my VSCode but I realised rules folder is empty and also why its empty, so should I directly clone the original capa repo and later form a different branch ??

**Comment by mr-tz** [comment]
Are your issues addressed in https://github.com/mandiant/capa/blob/master/doc/installation.md#method-3-inspecting-the-capa-source-code? If not we should update the documentation to clarify.

**Comment by tehami02** [comment]
@mr-tz Yeah, documentation says to cl9ne the origibal capa repo but I forked the repo and cloned it as usually thats how we do it.

**Comment by xusheng6** [comment]
> @mr-tz Yeah, documentation says to clone the original capa repo but I forked the repo and cloned it as usually thats how we do it.

Sounds like you are not cloning the submodules at the same time?

**Comment by tehami02** [comment]
@xusheng6 do i need to clone sub modules seperately ??

**Comment by xusheng6** [comment]
> @xusheng6 do i need to clone sub modules seperately ??

Try run "git submodule update --init --recursive"

**Comment by tehami02** [comment]
I m linking my forked version's rules folder link it says 404 not found which should not happen - https://github.com/tehami02/capa-rules

**Comment by xusheng6** [comment]
> I m linking my forked version's rules folder link it says 404 not found which should not happen - https://github.com/tehami02/capa-rules
> 
> 
> 
> And this is my forked version link - https://github.com/tehami02/capa

Sorry I do not quite get it. Could you please explain what you are trying to do, how you attempted to do it, and what errors you are getting

**Comment by tehami02** [comment]
@xusheng6 Basically I forked the capa repo but my fork does not contains files in rules folder as it contains in original one.

**Comment by tehami02** [comment]
@xusheng6 my only doubt is should I clone the original capa repo ??

**Comment by xusheng6** [comment]
> @xusheng6 my only doubt is should I clone the original capa repo ??

Sorry I am not sure

**Comment by tehami02** [comment]
@xusheng6 I saw pull requests all contributors are using forked repo and their repo contains all files properly ( for example - https://github.com/linpeiyu164/capa/tree/master) repo of a contributor, but I forked capa repo once agin from an another account ( link - https://github.com/testercse/capa) and found that it also has same problem folder like rules shows 404 error.

**Comment by tehami02** [comment]
Sorry for all the chaos, for now I cloned capa repo directly and issue is sorted, I have all the files locally.

**Comment by williballenthin** [comment]
glad you got this figured out @tehami02!

im going to go back and hide a bunch of the comments above so that this thread appears on-topic and related to the name demangling issue. feel free to continue to ask questions and we're here to help.
